src/BehavioralFsm.js

```diff
@@ -194,24 +194,41 @@ _.extend( BehavioralFsm.prototype, {
 	deferUntilTransition: function( client, stateName ) {
 		var clientMeta = this.ensureClientMeta( client );
 		var stateList = _.isArray( stateName ) ? stateName : ( stateName ? [ stateName ] : undefined );
+		var prom = null;
 		if ( clientMeta.currentActionArgs ) {
-			var queued = {
-				type: events.NEXT_TRANSITION,
-				untilState: stateList,
-				args: clientMeta.currentActionArgs
-			};
-			clientMeta.inputQueue.push( queued );
-			var eventPayload = this.buildEventPayload( client, {
-				state: clientMeta.state,
-				queuedArgs: queued
-			} );
-			this.emit( events.DEFERRED, eventPayload );
+			var addToQueue = ( function( callback ) {
+				var queued = {
+					type: events.NEXT_TRANSITION,
+					untilState: stateList,
+					args: clientMeta.currentActionArgs
+				};
+				if ( callback ) {
+					queued.callback = callback;
```

Comment:
This breaks existing tests for the defer* functionality because there's an extra property there. Is there any way in the test suite that we can ignore this property? I couldn't find anything in the should.js documentation, but I've never used that lib before so I may have missed something